mitiq/ddd/insertion.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2022 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Tools to determine slack windows in circuits and to insert DDD sequences."""
+
+import numpy as np
+from mitiq.interface import convert_to_mitiq
+from mitiq import QPROGRAM
+
+
+def get_circuit_mask(circuit: QPROGRAM) -> np.ndarray:
+    """Given a circuit with n qubits and d moments returns a matrix
+    A with n rows and d columns. The matrix elements are A_{i,j} = 1 if
+    there is a gate acting on qubit i at moment j, while A_{i,j} = 0 otherwise.
+    """
+    mitiq_circuit, _ = convert_to_mitiq(circuit)
+    qubits = sorted(mitiq_circuit.all_qubits())
```

Comment:
I know I suggested sorting in the RFC, but I am actually not sure. If it works without sorting, it is probably better to remove it. Otherwise we risk unexpected re-ordering of qubits. 
Do you agree?

Reply:
Good catch! So I agree on sorting. 
I noticed that a similar sorting is implicitly applied when calling print(circuit). So the mask that we get when sorting should match the printed circuit and this is good.